### PR TITLE
optimize: fall back to any of available cluster address when query cluster address is empty

### LINKED ISSUE #6790: removeOfflineAddressesIfNecessary method may cause the service list to be empty when switching clusters
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
removeOfflineAddressesIfNecessary method may cause the service list to be empty when switching clusters.
dce87acf96ae52599a79df177ed88579
我认为这里应该防推空,如果没有交集就不做任何处理,这样就可以解决问题了
I think it should be anti-empty here. If there is no intersection, we will not do any processing, so that the problem can be solved

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by laywin**
- [x] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

fall back to any of available cluster address when query cluster address is empty

### Ⅱ. Does this pull request fix one issue?

 fix: https://github.com/apache/incubator-seata/issues/6790

### CODE REVIEW

**Comment by laywin** [comment]
@funky-eyes  PTAL

File: discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java [hunk 1]
```diff
@@ -119,12 +120,28 @@ default String getServiceGroup(String key) {
...
+                k -> new ConcurrentHashMap<>());
+
+        String clusterName = getServiceGroup(transactionServiceGroup);
+        List<InetSocketAddress> inetSocketAddresses = clusterAddressMap.get(clusterName);
+        if (CollectionUtils.isNotEmpty(inetSocketAddresses)) {
+            return inetSocketAddresses;
+        }
+
+        // fall back to addresses of any cluster
+        return clusterAddressMap.values().stream().findAny().orElse(Collections.emptyList());
```
Comment (funky-eyes) [inline discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java]: findAny会不会找到一个emptyList？
Will findAny find an emptyList?

File: discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java [hunk 2]
```diff
@@ -137,15 +154,18 @@ default List<InetSocketAddress> refreshAliveLookup(String transactionServiceGrou
...
 
-        List<InetSocketAddress> currentAddresses = CURRENT_ADDRESS_MAP.getOrDefault(clusterName, Collections.emptyList());
+        List<InetSocketAddress> currentAddresses = clusterAddressMap.getOrDefault(clusterName, new ArrayList<>());
 
         List<InetSocketAddress> inetSocketAddresses = currentAddresses
                 .stream().filter(newAddressed::contains).collect(
                         Collectors.toList());
 
-        CURRENT_ADDRESS_MAP.put(clusterName, inetSocketAddresses);
+        clusterAddressMap.put(clusterName, inetSocketAddresses);
```
Comment (funky-eyes) [inline discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java]: 一定要保证inetSocketAddresses不为空再进行put
Be sure to ensure that inetSocketAddresses is not empty before putting.

File: discovery/seata-discovery-namingserver/src/main/java/org/apache/seata/discovery/registry/namingserver/NamingserverRegistryServiceImpl.java [hunk 3]
```diff
@@ -322,17 +321,6 @@ public void unsubscribe(String vGroup) throws Exception {
         isSubscribed = false;
     }
 
-    @Override
-    public List<InetSocketAddress> aliveLookup(String transactionServiceGroup) {
```
Comment (funky-eyes) [inline discovery/seata-discovery-namingserver/src/main/java/org/apache/seata/discovery/registry/namingserver/NamingserverRegistryServiceImpl.java:326]: 不要删除这块代码，namingserver设计与其他注册中心不同，namingserver自身存储cluster，缓存应该为group，group，list即可

File: discovery/seata-discovery-namingserver/src/main/java/org/apache/seata/discovery/registry/namingserver/NamingserverRegistryServiceImpl.java [hunk 4]
```diff
@@ -322,17 +321,6 @@ public void unsubscribe(String vGroup) throws Exception {
...
-        return CURRENT_ADDRESS_MAP.computeIfAbsent(transactionServiceGroup, k -> new ArrayList<>());
-    }
-
-    @Override
-    public List<InetSocketAddress> refreshAliveLookup(String transactionServiceGroup,
```
Comment (funky-eyes) [inline discovery/seata-discovery-namingserver/src/main/java/org/apache/seata/discovery/registry/namingserver/NamingserverRegistryServiceImpl.java:331]: 同上

File: discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java (hunk 1 above)
Comment (laywin) [reply discovery/seata-discovery-core/src/main/java/org/apache/seata/discovery/registry/RegistryService.java]: done

File: changes/en-us/2.x.md [hunk 5]
```diff
@@ -70,6 +70,7 @@ Add changes here for all PR submitted to the 2.x branch.
 - [[#6787](https://github.com/apache/incubator-seata/pull/6787)] upgrade elliptic to 6.5.7
 - [[#6783](https://github.com/apache/incubator-seata/pull/6783)] rename the server naming/v1 api to vgroup/v1
 - [[#6793](https://github.com/apache/incubator-seata/pull/6793)] fix npmjs conflicts
+- [[#6797](https://github.com/apache/incubator-seata/pull/6797)] fall back to any of available cluster address when query cluster address is empty
```
Comment (funky-eyes) [inline changes/en-us/2.x.md]: 应该登记到bug分类去，等优化成发布订阅模式才是属于优化
It should be registered in the bug classification. When it is optimized into the publish-subscribe mode, it is an optimization.

Comment (laywin) [reply changes/en-us/2.x.md]: done

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**